### ISSUE TITLE
add filtering/select function

### ISSUE THREAD

**Issue by jspaaks** [matchms/matchms-backup#54]
- [x] add a harmonization method, called ``select`` or ``slice`` or something to that effect, to ``harmonization/``. This methods lets users select a subset of masses and intensity. I think this is similar to ``max_frag``, ``min_frag``, ``min_loss`` and ``max_loss`` from @florian-huber's ``Spectrum`` implementation. ``select()`` could have default parameters like:

    ```
    spectrum = select(spectrum, min_frag=0.0, max_frag=1000.0,
                      min_loss=10.0, max_loss=200.0)
    ```
    The argument names are misleading btw, I think we should change them if we can. Maybe ``min_mz``, ``max_mz``, ``min_intensity``, ``max_intensity``? Currently the code uses 'losses' and 'peaks' interchangably, we should either
   - pick one and stick to it, or
   - avoid the issue altogether and use 'intensity' like I just did. 
- [x] add test

**Comment by jspaaks** [comment]
~~maybe add a harmonization function~~

~~spectrum = add_losses(spectrum, params)~~

~~(needs metadata, like precursor mass, and needs annotation 'peak' or 'loss')~~

**Comment by jspaaks** [comment]
This issue is addressed in PR #87.
